### PR TITLE
Adding previous price as reference point on pnl graph

### PR DESCRIPTION
**Pull request by IanWoodard**
Instead of having the PnL graph always be stagnant, this update sets the reference point to the price of the last modification and plots the current price on the graph.
Screenshot 2023-02-23 at 8 51 06 PM

### CODE REVIEW

File: prime/src/components/graph/PnLGraph.tsx [hunk 1]
```diff
@@ -252,23 +259,74 @@ export default function PnLGraph(props: PnLGraphProps) {
...
+          true,
+          activeChain
+        );
+      } catch (e) {
+        console.error(e);
+      }
+      if (etherscanResult == null || !Array.isArray(etherscanResult.data.result)) return [];
+      if (etherscanResult.data.result.length === 0) return [];
+      const mostRecentEvent = etherscanResult.data.result[0];
+      const mostRecentEventData: BigNumber = ethers.utils.defaultAbiCoder.decode(['uint256'], mostRecentEvent.data)[0];
```
Comment (haydenshively) [inline prime/src/components/graph/PnLGraph.tsx:285]: Does Etherscan promise to sort for us?

File: prime/src/components/graph/PnLGraph.tsx [hunk 2]
```diff
@@ -252,23 +259,74 @@ export default function PnLGraph(props: PnLGraphProps) {
...
+
+  // If we're showing hypothetical, we want to use the current price as the price at the last update
+  if (isShowingHypothetical) previousPrice = currentPrice;
+
+  if (inTermsOfToken0) {
+    currentPrice = 1 / currentPrice;
+    previousPrice = 1 / previousPrice;
+  }
+  const priceA = previousPrice / PLOT_X_SCALE;
+  const priceB = previousPrice * PLOT_X_SCALE;
```
Comment (haydenshively) [inline prime/src/components/graph/PnLGraph.tsx:322]: We might want to adjust these and/or `PLOT_X_SCALE` in order to force both currentPrice and previousPrice to be visible

File: prime/src/components/graph/PnLGraph.tsx [hunk 3]
```diff
@@ -370,7 +428,7 @@ export default function PnLGraph(props: PnLGraphProps) {
               />
               <YAxis stroke={SECONDARY_COLOR} fontSize='14px' />
               <ReferenceLine y={0} stroke={SECONDARY_COLOR} />
-              <ReferenceLine x={price} stroke={SECONDARY_COLOR} strokeWidth={2} />
+              <ReferenceLine x={currentPrice} stroke={SECONDARY_COLOR} strokeWidth={2} />
```
Comment (haydenshively) [inline prime/src/components/graph/PnLGraph.tsx:431]: Let's have an axis-colored reference line for `previousPrice`, and a bright white (or blue or something) reference line for `currentPrice`

**Review by haydenshively — COMMENTED** [review]
Few cosmetic changes, otherwise LGTM

File: prime/src/components/graph/PnLGraph.tsx [hunk 4]
```diff
@@ -252,23 +259,77 @@ export default function PnLGraph(props: PnLGraphProps) {
...
+          ALOE_II_FRONTEND_MANAGER_ADDRESS,
+          [TOPIC0_MODIFY_EVENT, `0x000000000000000000000000${marginAccount.address.slice(2)}`],
+          true,
+          activeChain
+        );
+      } catch (e) {
+        console.error(e);
+      }
+      if (etherscanResult == null || !Array.isArray(etherscanResult.data.result)) return [];
+      if (etherscanResult.data.result.length === 0) return [];
```
Comment (haydenshively) [inline prime/src/components/graph/PnLGraph.tsx:283]: Why return empty array here?

File: prime/src/components/graph/PnLGraph.tsx [hunk 5]
```diff
@@ -252,23 +259,77 @@ export default function PnLGraph(props: PnLGraphProps) {
...
+
+  // If we're showing hypothetical, we want to use the current price as the price at the last update
+  if (isShowingHypothetical) previousPrice = currentPrice;
+
+  if (inTermsOfToken0) {
+    currentPrice = 1 / currentPrice;
+    previousPrice = 1 / previousPrice;
+  }
+  const priceA = Math.min(currentPrice, previousPrice) / PLOT_X_SCALE;
+  const priceB = Math.max(currentPrice, previousPrice) * PLOT_X_SCALE;
```
Comment (haydenshively) [inline prime/src/components/graph/PnLGraph.tsx:325]: 👍

**Review by haydenshively — APPROVED** [review]

File: prime/src/components/graph/PnLGraph.tsx (hunk 4 above)
Comment (IanWoodard) [reply prime/src/components/graph/PnLGraph.tsx:283]: Since if there are no events, we don't want to try to find the most recent

**Pull request merged**